yt_dlp/extractor/sverigesradio.py

```diff
@@ -21,7 +26,13 @@ class SverigesRadioBaseIE(InfoExtractor):
     }
 
     def _real_extract(self, url):
-        audio_id = self._match_id(url)
+        audio_id, display_id = self._match_valid_url(url).group('id', 'slug')
+        if not audio_id:
+            webpage = self._download_webpage(url, display_id)
+            button = extract_attributes(get_element_html_by_class('audio-button', webpage) or '')
+            audio_id = traverse_obj(button, 'data-audio-id', 'data-publication-id') or self._parse_json(
+                get_element_by_id('gtm-metadata', webpage), display_id)['pageId']
```

Comment:
Cosmetic, but I like having the `or` branch clearly separated. inlining `extract_attributes` could be done either way, since it gets `try_call`ed we need not `or ''`:
```suggestion
            audio_id = (
                traverse_obj(
                    get_element_html_by_class('audio-button', webpage),
                    ({extract_attributes}, ('data-audio-id', 'data-publication-id')), get_all=False)
                or self._parse_json(get_element_by_id('gtm-metadata', webpage), display_id)['pageId'])
```